pilot/pkg/security/authz/model/model.go

```diff
@@ -136,33 +137,36 @@ func NewModel(role *istio_rbac.ServiceRole, bindings []*istio_rbac.ServiceRoleBi
 
 	for _, binding := range bindings {
 		for _, subject := range binding.Subjects {
-			var principal Principal
-			principal.User = subject.User
-			principal.Names = subject.Names
-			principal.NotNames = subject.NotNames
-			principal.Groups = subject.Groups
-			principal.Group = subject.Group
-			principal.NotGroups = subject.NotGroups
-			principal.Namespaces = subject.Namespaces
-			principal.NotNamespaces = subject.NotNamespaces
-			principal.IPs = subject.Ips
-			principal.NotIPs = subject.NotIps
-
-			property := KeyValues{}
-			for k, v := range subject.Properties {
-				property[k] = []string{v}
+			users := getPrincipalsIncludingAliases(trustDomainAliases, []string{subject.User})
+			for _, user := range users {
+				var principal Principal
+				principal.User = user
```

Comment:
Sounds good. Thanks!

Reply:
Actually, are you worried about performance? I feel like having 1 to 1 mapping with the Istio authz API makes it easier to understand.

Reply:
Also, if we're not putting a for loop here, we'll need to put it somewhere else (i.e. the part from model -> envoy config), since `principals` in v2 and `user` in v1 are treated slightly different?

Reply:
This is for both performance, simplicity and consistent with the v1beta1 generation. This should be put inside the principal generation, the same way as the v1beta1 code.

Also I think it doesn't make the authz API easier to understand but actually opposite it makes the generated filter config harder to understand because you have to repeat other fields everywhere.

Reply:
Would it be better if we set `principal.Names = users`? https://github.com/istio/istio/blob/master/pilot/pkg/security/authz/model/model.go#L141 and don't use `principal.User` anymore?

Reply:
Yeah, I think that is also okay because the `subject.names` is never released in the user-facing API so we can just change it to `subject.users`, you may need to update the tests for this change though.

Reply:
I changed `user` to `users` instead. PTAL. Thanks!